test/tests/parallel/set3/dovecot_inet_protocol.bats

```diff
@@ -0,0 +1,47 @@
+TEST_NAME_PREFIX='Dovecot protocols:'
+CONTAINER_NAME='dummy'
+CONTAINER_NAME_ALL='dms-test-dovecot_protocols_all'
+CONTAINER_NAME_IPV4='dms-test-dovecot_protocols_ipv4'
+CONTAINER_NAME_IPV6='dms-test-dovecot_protocols_ipv6'
+
+load "${REPOSITORY_ROOT}/test/helper/setup"
+load "${REPOSITORY_ROOT}/test/helper/common"
+
+function setup_file() {
+  CONTAINER_NAME=${CONTAINER_NAME_ALL}
+  init_with_defaults
+  local CUSTOM_SETUP_ARGUMENTS=(--env DOVECOT_INET_PROTOCOLS=)
+  common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
+
+  CONTAINER_NAME=${CONTAINER_NAME_IPV4}
+  init_with_defaults
+  local CUSTOM_SETUP_ARGUMENTS=(--env DOVECOT_INET_PROTOCOLS=ipv4)
+  common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
+
+  CONTAINER_NAME=${CONTAINER_NAME_IPV6}
+  init_with_defaults
+  local CUSTOM_SETUP_ARGUMENTS=(--env DOVECOT_INET_PROTOCOLS=ipv6)
+  common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
+}
```

Comment:
Not required for the PR, but `setup_file()` preps setup for one or more tests, but as there is only 1 test for each, technically we could have each `@test` inline the relevant setup block from here, reducing resource requirements and time to first test running.

Alternatively in future it could be split into 3 sub-test files too I suppose that can all be run at once if desired :man_shrugging:

Reply:
Agreed, should be in another PR. Not sure what to do now though: resolve or leave unresolved?

Reply:
Can leave as-is, and this feedback can remain unresolved so it's easier to notice this needs to be resolved in a follow-up PR? (_although @casperklein isn't fond of those AFAIK, so if he'd prefer we tackle it now that's fine_)

Reply:
You're right, @casperklein does indeed not like it. I decided to put it into this PR and inlined container creation.